admin/code/CampaignAdmin.php

```diff
@@ -0,0 +1,408 @@
+<?php
+
+/**
+ * Campaign section of the CMS
+ *
+ * @package framework
+ * @subpackage admin
+ */
+class CampaignAdmin extends LeftAndMain implements PermissionProvider {
+
+	private static $allowed_actions = [
+		'set',
+		'sets',
+		'schema',
+		'DetailEditForm',
+		'readCampaigns',
+		'createCampaign',
+		'readCampaign',
+		'updateCampaign',
+		'deleteCampaign',
+	];
+
+	private static $menu_priority = 11;
+
+	private static $menu_title = 'Campaigns';
+
+	private static $url_handlers = [
+		'GET sets' => 'readCampaigns',
+		'POST set/$ID' => 'createCampaign',
+		'GET set/$ID' => 'readCampaign',
+		'PUT set/$ID' => 'updateCampaign',
+		'DELETE set/$ID' => 'deleteCampaign',
+	];
+
+	private static $url_segment = 'campaigns';
+
+	public function getClientConfig() {
+		return array_merge(parent::getClientConfig(), [
+			'forms' => [
+				// TODO Use schemaUrl instead
+				'editForm' => [
+					'schemaUrl' => $this->Link('schema/EditForm')
+				]
+			]
+		]);
+	}
+
+	public function schema($request) {
+		// TODO Hardcoding schema until we can get GridField to generate a schema dynamically
+		$json = <<<JSON
```

Comment:
Fairly minor, but slightly smelly to have JSON as a big string here. There doesn't seem to be any interpolation here, so this could be made either by json_encode'ing a PHP array structure, or by having a admin/code/CampaignAdmin.schema.json file. But if going away soon, not a critical fix.